src/compute/aggregate/sum.rs

```diff
@@ -23,19 +23,20 @@ pub trait Sum<T> {
 #[clone(target = "x86_64+avx")]
 fn nonnull_sum<T>(values: &[T]) -> T
 where
-    T: NativeType + Simd,
-    T::Simd: Add<Output = T::Simd> + Sum<T>,
+    T: NativeType + Simd + Add<Output = T> + std::iter::Sum<T>,
+    T::Simd: Sum<T> + Add<Output = T::Simd>,
 {
-    let mut chunks = values.chunks_exact(T::Simd::LANES);
-
-    let sum = chunks.by_ref().fold(T::Simd::default(), |acc, chunk| {
-        acc + T::Simd::from_chunk(chunk)
-    });
-
-    let remainder = T::Simd::from_incomplete_chunk(chunks.remainder(), T::default());
-    let reduced = sum + remainder;
+    // Safety:
+    // T::Simd is the vector type T and the alignment is similar to aligning to [T; alignment]
+    // the alignment of T::Simd ensures that it fits T.
+    let (head, simd_vals, tail) = unsafe { values.align_to::<T::Simd>() };
+
+    let mut reduced = T::Simd::from_incomplete_chunk(&[], T::default());
+    for chunk in simd_vals {
+        reduced = reduced + chunk.clone()
```

Comment:
why is this clone necessary? `chunks` should all be `Copy` and thus `*chunk` should be enough here.

Reply:
That is indeed the case for the `packed_simd` vector lane, but the core version does not implement copy. :/

Reply:
We can implement it, right?  =)

```rust
#[derive(Copy, Clone)]
pub struct $name(pub [$type; $lanes]);
```

and

```
pub trait NativeSimd: Default + Copy
```

Reply:
Oh.. I did not realize we create the native types. I thought the were from `core::arch`. My bad!